dht.go

```diff
@@ -251,25 +253,34 @@ func makeDHT(ctx context.Context, h host.Host, cfg config) (*IpfsDHT, error) {
 	return dht, nil
 }
 
-func makeRoutingTable(h host.Host, cfg config) (*kb.RoutingTable, error) {
-	self := kb.ConvertPeerID(h.ID())
+func makeRoutingTable(dht *IpfsDHT, cfg config) (*kb.RoutingTable, error) {
+	self := kb.ConvertPeerID(dht.host.ID())
 	// construct the routing table with a peer validation function
 	pvF := func(c context.Context, p peer.ID) bool {
```

Comment:
this change looks fine to me.

At some point, we should declare the `pvF` as a normal method of the DHT, rather than creating it as an anonymous function in `makeRoutingTable`, which ought to simplify some of these references.